syntax = "proto3";

package roles;

service Roles {
    rpc GetUserRole(GetUserRoleRequest) returns (GetUserRoleResponse);
}

enum UserRole {
    CUSTOMER = 0;
    COURIER = 1;
    RESTAURANT_MANAGER = 2;
    MODERATOR = 3;
}

message GetUserRoleRequest {
    string access_token = 1;
}

message GetUserRoleResponse {
    int64 user_id = 1;
    UserRole role = 2;
}